server/src/main/java/io/crate/execution/dml/TranslogIndexer.java

```diff
@@ -138,7 +147,27 @@ private IndexDocumentBuilder populateLuceneFields(BytesReference source) throws
             entry.setValue(castValue);
         }
 
-        addIndexColumns(indexColumns, docBuilder);
+        for (var indexColumn : indexColumns) {
```

Comment:
I'd pull this into a separate method, as otherwise `populateLuceneFields` gets a bit too big to easily read.